runtime/runtime.h

```diff
@@ -5,13 +5,25 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <stdio.h>
-#include "ocl.h"
-#include "time_utils.h"
 
 typedef struct ContextImpl* Context;
 typedef struct KernelImpl* Kernel;
 typedef struct BufferImpl* Buffer;
 
+typedef struct {
+  size_t size;
+  void* value;
+} KernelArg;
+
+#ifndef SHINE_TARGET_GAP8
+    #include "ocl/ocl.h"
+#else
+    #include "gap8/gap8.h"
+#endif
+#include "time_utils.h"
```

Comment:
I would rather have the backend-specific headers include the generic header, and the user directly include the backend-specific header (instead of including the generic one and having to define a macro to instantiate it).

Reply:
Right, I just followed the current paradigm, but backend-specific headers including generic header sound better for me as well :+1: